samza-core/src/main/java/org/apache/samza/storage/ChangelogStreamManager.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JavaStorageConfig;
+import org.apache.samza.config.JavaSystemConfig;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.coordinator.stream.messages.CoordinatorStreamMessage;
+import org.apache.samza.coordinator.stream.messages.SetChangelogMapping;
+import org.apache.samza.coordinator.stream.CoordinatorStreamManager;
+import org.apache.samza.system.StreamSpec;
+import org.apache.samza.system.SystemAdmin;
+import org.apache.samza.system.SystemStream;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The Changelog manager is used to persist and read the changelog information from the coordinator stream.
+ */
+public class ChangelogStreamManager {
+
+  private static final Logger log = LoggerFactory.getLogger(ChangelogStreamManager.class);
+  private final CoordinatorStreamManager coordinatorStreamManager;
+
+  // This is legacy for changelog. Need to investigate what happens if you use a different source name
+  private static final String SOURCE = "JobModelManager";
+
+  /**
+   * Construct Changelog manager without a coordinator stream.
+   *
+   *  @param source Name of caller.
+   */
+  public ChangelogStreamManager(String source) {
+    coordinatorStreamManager = null;
```

Comment:
How does this work? Where is this used?

Reply:
Since this's the only state in this class, this would imply that whatever methods ZkJobCoordinator is using can be made static. Might be cleaner than adding this constructor.

Reply:
Agreed, will change that.